packages/host/app/resources/card-type.ts

```diff
@@ -203,3 +205,34 @@ export function getCardType(parent: object, card: () => typeof BaseDef) {
     },
   })) as CardType;
 }
+
+function isCodeRefType(type: any): type is CodeRefType {
+  return type && isCodeRef(type) && 'displayName' in type;
+}
+
+export function isFieldOfType(obj: any): obj is FieldOfType {
+  return obj && 'card' in obj;
+}
+
+export function codeRefName(f: Type | FieldOfType) {
+  let codeRef: CodeRef;
+
+  if (isFieldOfType(f)) {
+    // if field
+    codeRef = isCodeRefType(f.card) ? f.card : f.card.codeRef;
+    if (isResolvedCodeRef(codeRef)) {
+      return codeRef.name;
+    }
+  } else {
+    // if ancestor
+    codeRef = f.codeRef;
+    if (isResolvedCodeRef(codeRef)) {
+      return codeRef.name;
+    } else {
+      if (codeRef.type === 'ancestorOf') {
+        return f.declarationName; //local card or field
+      }
+    }
+  }
+  return;
+}
```

Comment:
I feel that this is such a confusing function. Thats purpose is to extract a codeRefName that will be used for comparison. We want a key that can compare declarations of a previous module with the current one so when opening definition, we highlight the correct one.

Reply:
Why we added `.declarationName` is because for a file with a locally defined ancestor. We do not store name. If its a class, the name. We kinda need this to make the comparison.